app/src/main/java/edu/mills/findeatfood/MainActivity.java

```diff
@@ -1,13 +1,314 @@
+        for (int i = 0; i < dietaryOptionsCode.length; i++) {
+            restrictionsHashMap.put(dietaryOptionsName[i], dietaryOptionsCode[i]);
+        }
+
+        for (int i = 0; i < allergyOptionsCode.length; i++) {
+            restrictionsHashMap.put(allergyOptionsName[i], allergyOptionsCode[i]);
+        }
+
+        int dietCheckBoxes =
+                getResources().getStringArray(R.array.dietOptionsName).length;
+        int allergyCheckBoxes =
+                getResources().getStringArray(R.array.allergyOptionsName).length;
+        LinearLayout dietaryOptions = (LinearLayout) findViewById(R.id.dietaryOptionsLayout);
+        ArrayList<String> dietRestrictions = new ArrayList<String>();
+        ArrayList<String> allergyRestrictions = new ArrayList<String>();
+
+        for (int i = 0; i < dietCheckBoxes; i++) {
+            CheckBox cb =
+                    (CheckBox) dietaryOptions.findViewById(dietOptionsIds.get(i));
+            if (cb != null && cb.isChecked()) {
+                String cbName = cb.getText().toString();
+                String cbCode = restrictionsHashMap.get(cbName);
+                dietRestrictions.add(cbCode);
+            }
+        }
+
+        for (int j = 0; j < allergyCheckBoxes; j++) {
+            CheckBox cb =
+                    (CheckBox) dietaryOptions.findViewById(allergyOptionsIds.get(j));
+            if (cb != null && cb.isChecked()) {
+                String cbName = cb.getText().toString();
+                String cbCode = restrictionsHashMap.get(cbName);
+                allergyRestrictions.add(cbCode);
+            }
+        }
+
+        Bundle recipeBundle = new Bundle();
+        recipeBundle.putStringArray(INGREDIENTS, ingredients);
+        recipeBundle.putStringArrayList(DIET_RESTRICTIONS, dietRestrictions);
+        recipeBundle.putStringArrayList(ALLERGY_RESTRICTIONS, allergyRestrictions);
+        ResultsFragment resultsFrag = new ResultsFragment();
+        resultsFrag.setArguments(recipeBundle);
+        doFragTransaction(resultsFrag);
+    }
+
+    public void onEditClicked(View v) {
+        IngredientsFragment ingredientsFrag = new IngredientsFragment();
+        doFragTransaction(ingredientsFrag);
+    }
+
+    public void onDietaryClicked(View v) {
+        EditText addIngredientET = (EditText) findViewById(R.id.addIngredientET);
+        if (addIngredientET.getText().toString().equals("")) {
+            Toast.makeText(getApplicationContext(), R.string.error_ingredient, Toast.LENGTH_SHORT).show();
+        } else {
+            DietaryFragment dietaryFrag = new DietaryFragment();
+            doFragTransaction(dietaryFrag);
+            setIngredients(parseIngredientInput(addIngredientET.getText().toString()));
+        }
+    }
+
+    private void setIngredients(String[] ingredients) {
+        this.ingredients = ingredients;
+    }
+
+    private String[] parseIngredientInput(String ingredients) {
+        return ingredients.split("\\s*,\\s*");
```

Comment:
I like to have a comment (not javadoc) before a regular expression is used, such as: "Split the ingredients along commas with optional surrounding white space."